### PR TITLE
make 404-page better than ever before

### LINKED ISSUE #37: Build proper 404 page
The 404 page should show the generic layout and have some basic content. It should also determine the locale from the URL and show navigation for the correct language.

### PR DESCRIPTION
**Pull request by kahlstrm**
Improve 404 page not found page with bad localization support and a link to homepage.

Relies on typography from #48.

Closes #37.

### CODE REVIEW

**Review by tuukka-miettinen — APPROVED** [review]
Could also create an issue for the nearest-matching path from strapi link in 404 page

File: src/pages/404.tsx [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
-    <main>
-      <h1>Not found</h1>
-      <p>{path}</p>
-    </main>
+    <PageContext.Provider value={context}>
+      <Layout>
+        <SideBar>
+          {locale === "fi" ? (
+            <>
+              <h1>Sivua {pathname} ei löytynyt</h1>
```
Comment (PurkkaKoodari) [inline src/pages/404.tsx]: We should change this to `<H1>` from typography.tsx when merging (or rebase/merge it to be so beforehand)

**Review by kahlstrm — COMMENTED** [review]
Very good very nice

**Review by PurkkaKoodari — APPROVED** [review]
"Very good very nice" --@kqlski

**Comment by PurkkaKoodari** [comment]
Minor issue: the 404.html page is rendered with its in-filesystem pathname, which flashes before the page hydrates. Maybe this would be better without the pathname, then.

image

**Pull request merged**